lib/product.service.ts

```diff
@@ -114,5 +165,93 @@ export class ProductStack extends Stack {
       allowMethods: ['GET', 'OPTIONS'],
```

Comment:
a common method could be created to add the corsPreflight as it is duplicated. YOu can jus send the api resource as the method param.